pkg/pulumiyaml/analyser.go

```diff
@@ -1096,6 +1096,15 @@ func (tc *typeCache) typeConfig(r *Runner, node configNode) bool {
 	return true
 }
 
+func (tc *typeCache) compatibleType(a, b schema.Type) bool {
```

Comment:
I would rename this method to `isConfigEquivalent` or something similar. Its not clear without context what "compatibleType" means.

Reply:
Is there a downside to defaulting integer -> number, or do we want to keep it strict?

Reply:
There is a downside (we could allow programs that the user thinks we will reject), but it's minimal. I support allowing `integer -> number`, but not the other way around.